storage-proofs/src/circuit/porc.rs

```diff
@@ -223,7 +223,7 @@ impl<'a, E: JubjubEngine> Circuit<E> for PoRCCircuit<'a, E> {
                 // Compute the new subtree value
                 cur = pedersen_hash::pedersen_hash(
                     cs.namespace(|| "computation of pedersen hash"),
-                    pedersen_hash::Personalization::MerkleTree(i),
+                    pedersen_hash::Personalization::NoteCommitment,
```

Comment:
I always thought we need this, for the same reason we need this to avoid 2nd preimage attacks in the regular merkle tree building. Could you explain to me why that is not the case?

Reply:
I'm going to try to reinstate it just to minimize change.

However, I think the better question is why we *do* need it. I think the 'attack' this prevents is that without it a miner can arrange for collisions of `CommD` computed over sectors of different sizes (by using hashes from an intermediate row as leaves of a smaller sector).

Reply:
> I think the better question is why we _do_ need it.

I agree, I just would like to be sure that we don't need it before ripping it out.

Reply:
After I reworked everything, the initial awkwardness became tractable so I backed this change out. In case it comes up again, I still think we can live without the personalization as long as we don't try to distinguish different-sized sectors by their data commitment (which we should not do).